### PR TITLE
S3 destination: updating specs regarding certification

### LINKED ISSUE #11835: S3 destination: update spec regarding certification checklist
All fields in spec should have the following:

* required fields should be marked with an asterisk (eg. "Authentication *");
* optional fields should be appended with "(Optional)" (eg. "End Date (Optional)" )

Input fields should have following order:

1. Name *
1. Authentication *
1. All other required fields
1. Any optional fields

┆Issue is synchronized with this [Monday item](https://airbyte-bunch.monday.com/boards/2536787439/pulses/2537082891) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sashaNeshcheret**
## What
We updated specs regarding certification

## How
We added * or 'Optional' to fields names, reordered fields.

## Recommended reading order
1. `x.json`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by sashaNeshcheret** [comment]
@misteryeo, could you take a look.

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 1]
```diff
@@ -16,33 +16,34 @@
...
-        "examples": ["http://localhost:9000"],
+        "description": "The access key id to access the S3 bucket. Airbyte requires Read and Write permissions to the given bucket.",
+        "title": "S3 Key Id *",
+        "airbyte_secret": true,
+        "examples": ["A012345678910EXAMPLE"],
         "order": 0
       },
+      "secret_access_key": {
+        "type": "string",
+        "description": "The corresponding secret to the access key id",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:29]:
```suggestion
        "description": "The corresponding secret to the access key ID",
```

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 2]
```diff
@@ -16,33 +16,34 @@
     ],
     "additionalProperties": false,
     "properties": {
-      "s3_endpoint": {
-        "title": "Endpoint",
+      "access_key_id": {
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:19]: For all of these fields, where are the relevant docs linked so a user can find more information as part of this checklist item?

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 3]
```diff
@@ -211,14 +198,16 @@
                       }
                     }
                   }
-                ]
+                ],
+                "order": 1
               },
               "part_size_mb": {
-                "title": "Block Size (MB) for Amazon S3 multipart upload",
+                "title": "dddBlock Size (MB) for Amazon S3 multipart upload (Optional)",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:205]: I think you added some random 'd's to this description?

**Review by misteryeo — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json (hunk 3 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:205]: Seems yes, it was attempt to recognize whether specs is reloaded.

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 4]
```diff
@@ -348,42 +340,59 @@
...
             }
           }
         ],
+        "order": 5
+      },
+      "s3_endpoint": {
+        "title": "Endpoint (Optional)",
+        "type": "string",
+        "default": "",
+        "description": "This is your S3 endpoint url.(if you are working with AWS S3, just leave empty).",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:385]:
```suggestion
        "description": "This is your S3 endpoint url. (If you are working with AWS S3, you can leave blank).",
```

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 5]
```diff
@@ -348,42 +340,59 @@
...
+        "examples": ["http://localhost:9000"],
+        "order": 6
+      },
+      "s3_path_format": {
+        "title": "s3_path_format (Optional)",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:390]:
```suggestion
        "title": "S3 Path Format (Optional)",
```

Please make sure all these fields are updated to reflect the name and not the code.

**Review by misteryeo — CHANGES_REQUESTED** [review]
Still don't see relevant docs linked in descriptions and left a few more comments.

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json (hunk 2 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:19]: Added links for required and optional fields.

File: airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json [hunk 6]
```diff
@@ -80,38 +81,24 @@
...
-        "airbyte_secret": true,
-        "examples": ["a012345678910ABCDEFGH/AbCdEfGhEXAMPLEKEY"],
-        "order": 6
-      },
       "format": {
-        "title": "Output Format",
+        "title": "Output Format *",
         "type": "object",
-        "description": "Output data format",
+        "description": "Output data format See [here](https://docs.airbyte.com/integrations/destinations/s3/#output-schema) for more details",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json:87]:
```suggestion
        "description": "Format of the data output. See [here](https://docs.airbyte.com/integrations/destinations/s3/#output-schema) for more details",
```

**Review by misteryeo — APPROVED** [review]
Just one suggestion otherwise good to go!

**Pull request merged**